shell/platform/android/io/flutter/app/FlutterPluginRegistry.java

```diff
@@ -72,11 +72,11 @@ public Registrar registrarFor(String pluginKey) {
     public void attach(FlutterView flutterView, Activity activity) {
         mFlutterView = flutterView;
         mActivity = activity;
-        mPlatformViewsController.attachFlutterView(flutterView);
+        mPlatformViewsController.attach(activity, flutterView, flutterView);
```

Comment:
So `FlutterView` happens to implement or own all of these params at once right now, which takes care of any potential dependent state that they may have on each other. If there's underlying setup work that needs to be done to relate these three together, then now it's possible to break `PlatformViewsController` by failing to initialize these with the right relationships to each other before passing them in. Ideally I think `attach()` would want to avoid potentially breaking based on dependencies across its params. Or at least make it clear in the method's documentation how these things need to be set up in relationship to each other before they're passed in, if that's totally unavoidable.

It looks like `TextureRegistry` probably needs its `SurfaceTexture` to exist within `Context`? I'm not seeing any blatant dependencies between `BinaryMessenger` and the other two reading through the interfaces, it looks like it may be totally independent. But if that's the case, it seems strange that it's being set and unset simultaneously with the other two instead of just set once in the constructor.

Reply:
Good point, though I think we can say that from `PlatformViewsController`'s perspective these are completely independent:

`BinaryMessenger` is the communication channel with the "Flutter app" (the Dart code), the Flutter app instance that is on the other side of the messenger is the one who's controlling the views managed by this controller (create, dispose, size, etc...)

`TextureRegistry` is the provider of textures into which the embedded views will be rendered, I don't think there's a dependency between `SurfaceTexture` and `Context`. A `SurfaceTexture` is attached to a GL context (which is different from `android.content.Context`).

`Context` is essentially the embedded view's connection to the world, e.g if a `FooMapView` needs to load resources it will get to them through this context.

I think we can even come up with a valid strange use case where these parameters does not maintain the same relationship they have right now, like: an Android app running 2 different "Flutter apps", and one of them is controlling an embedded view that is shown in the other Flutter app.

I'm not sure I see how can `PlatformViewsController` break as long as each of these parameters is valid on its own.

Reply:
Sounds good, thanks for the explanation!